### DISCUSSION TITLE
Mise à niveau - Erreur

### DISCUSSION THREAD

**Discussion by etienne-64** [dbarzin/mercator#498] [Q&A]
Bonjour,

Je rencontre une erreur au niveau de la mise à jour à l'issue du "sudo -u www-data composer update" (capture ci-dessous)

![image](https://github.com/dbarzin/mercator/assets/10651729/dcd12217-86e2-4dc6-a2d4-e4bdd98ea411)

Merci votre aide.

Stéphane

**Comment by dbarzin** [comment]
On en a discuté ici : https://github.com/dbarzin/mercator/issues/381

**Comment by etienne-64** [comment]
Merci pour ce retour très rapide.

C'est reparti mais une petite erreur quand je vais dans le menu Configuration > Rôles

![image](https://github.com/dbarzin/mercator/assets/10651729/94c3d7b9-4c6e-457d-b24d-bfac74d71b79)

**Comment by dbarzin** [comment]
Tu n'aurais pas oublié de migrer la base de données ?
    
    php artisan migrate

**Reply by etienne-64** [reply]
Non. J'ai également rejoué la séquence de réparation ( Repair the problems of migration ).
Le message en entier.
![image](https://github.com/dbarzin/mercator/assets/10651729/e728888b-2de0-405a-ab77-99827098543b)

**Reply by etienne-64** [reply]
Je l'ai rejoué "au cas où" ;)
/var/www/mercator# sudo -u www-data php artisan migrate 

   INFO  Nothing to migrate.  

Je suis en mariadb pas en mysql ... parfois des surprises mais pas à ce niveau.

**Reply by etienne-64** [reply]
J'ai modifié "à la main" (pas bien) le code et ça roule ;)

**Reply by dbarzin** [reply]
Tu peux partager la modification ?

**Reply by etienne-64** [reply]
Message d'erreur > "pas de roles.title dans le group by donc ...

```php
// Dans app/Http/Controllers/Admin/RolesController.php, ligne 63
// Remplacer
->groupBy('roles.id')
//Par 
->groupBy('roles.id','roles.title')
```

**Reply by etienne-64** [reply]
Un peu de transpiration pour cette mise à jour ;) mais elle en vaut le peine. Merci pour cette belle réalisation.

**Reply by dbarzin** [reply]
Merci, c'est corrigé.
